### DISCUSSION TITLE
Difference between stale-while-revalidate cache control header with and without value

### DISCUSSION THREAD

**Discussion by lakbychance** [vercel/vercel#7882] [Help] [answered]
`stale-while-revalidate` usage is defined without a value [here](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching)
 
<img width="802" alt="Screenshot 2022-05-27 at 12 21 09 PM" src="https://user-images.githubusercontent.com/29139815/170647815-ae8c5c3d-e061-480d-adfe-f5089a917d79.png">

But [here](https://vercel.com/docs/concepts/edge-network/caching) `stale-while-revalidate` usage is defined with the value and matches the general [spec](https://datatracker.ietf.org/doc/html/rfc5861#section-3)
<img width="715" alt="Screenshot 2022-05-27 at 12 21 41 PM" src="https://user-images.githubusercontent.com/29139815/170647964-bf43abe2-c98f-47ad-83e8-0ec65f9612fa.png">

What's the difference between both and use cases ?
* Does the first one mean that the cache can stay in stale state for a considerable amount of time and a blocking request to origin server is never made ? For eg - `s-maxage=1 stale-while-revalidate` means that post `1s`, vercel will try to revalidate the cache but there is no time limit to it and the user can be served a stale page for longer ?
* I think the second one ensures a blocking request to origin server in case the cache didn't get revalidated within **swr window** time and a new request came in. In case, it got revalidated then no request to origin server and cache is fresh for s-maxage and stale for further **swr window**

**Comment by leerob** [comment, marked as answer]
`stale-while-revalidate` with no defined duration is equivalent to “infinity” but as with any cache, there’s no guarantees on how long it stays in the cache.

**Reply by leerob** [reply]
I will make sure to update the docs here as well, too. Thank you!

**Reply by NobodysNightmare** [reply]
> stale-while-revalidate with no defined duration is equivalent to “infinity”

Is this the case? I found this discussion just now, but created an issue a few days ago about the missing value, see https://github.com/vercel/next.js/issues/51823

**tl; dr** from the issue: I can't find any formal specifications that mention that leaving out the duration is even allowed. All informal sources mention a duration.
